### PR TITLE
node and ci: add more linters

### PR DESCRIPTION
**Pull request by tbjump**
Changes: 
* update golangci-lint to latest version
* add golang linters: forcetypeassert, noctx, unparam, prealloc as recommended by Trail of Bits and because they seem useful
* fix all linter warnings

Todo: 
* some linter warnings were ignored by adding `//nolint ... TODO FIXME`. These spots should be revisited by the respective codeowners in the future because there may be low-impact bugs there.

### CODE REVIEW

**Review by claudijd — DISMISSED** [review]

File: .golangci.yml [hunk 1]
```diff
@@ -5,12 +5,20 @@ linters:
...
     - ineffassign
     - staticcheck
     - typecheck
     - unused
-    - gosec
 
+  # additional linters
+    - gosec
+    - forcetypeassert
+    # - contextcheck # disabled because
```
[the remark this answers is not in the document]
Comment (tbjump) [reply .golangci.yml:20]: fixed

File: node/hack/repair_eth/repair_eth.go [hunk 2]
```diff
@@ -1,3 +1,4 @@
+//nolint:forcetypeassert //this is a hack
```
[the remark this answers is not in the document]
Comment (tbjump) [reply node/hack/repair_eth/repair_eth.go:1]: 🤠

File: node/hack/repair_solana/repair.go [hunk 3]
```diff
@@ -266,11 +266,13 @@ func process(txRpc *rpc.TransactionWithMeta) (*solana.PublicKey, error) {
 	program, err := solana.PublicKeyFromBase58(*solanaAddr)
 	if err != nil {
 		log.Fatalf("Invalid program address: %v", err)
+		return nil, err
 	}
 
 	tx, err := txRpc.GetTransaction()
 	if err != nil {
 		log.Fatalf("Failed to unmarshal transaction: %v", err)
+		return nil, err
```
[the remark this answers is not in the document]
Comment (tbjump) [reply node/hack/repair_solana/repair.go:275]: I think it doesn't make a big difference if we add these useless returns vs. a comment to disable the linter. This is in the `hack/` dir anyways and I don't think anyone cares about code quality there.

File: node/pkg/governor/governor_prices.go [hunk 4]
```diff
@@ -156,8 +156,8 @@ func (gov *ChainGovernor) queryCoinGecko() error {
 		if exists {
 			// If a price is not set in CoinGecko, they return an empty entry. Treat that as a zero price.
 			price := float64(0)
-			m := data.(map[string]interface{})
-			if len(m) != 0 {
+			m, ok := data.(map[string]interface{})
+			if ok && len(m) != 0 {
```
[the remark this answers is not in the document]
Comment (tbjump) [reply node/pkg/governor/governor_prices.go]: added logging

**Review by hendrikhofstadt — APPROVED** [review]
Nice cleanups! Any more linters that would make sense adding going forward?

**Comment by tbjump** [comment]
Yes, there are a few more linters that I'd like to try out. https://golangci-lint.run/usage/linters/ has many more available for easy integration. But one step at a time.

File: node/pkg/accountant/accountant_test.go [hunk 5]
```diff
@@ -132,7 +132,7 @@ func newAccountantForTest(
 }
 
 // Converts a string into a go-ethereum Hash object used as test input.
-func hashFromString(str string) ethCommon.Hash {
+func hashFromString(str string) ethCommon.Hash { //nolint:unparam
```
Comment (bruce-riley) [inline node/pkg/accountant/accountant_test.go:135]: Why is this necessary?

**Review by bruce-riley — APPROVED** [review]

**Review by SEJeff — APPROVED** [review]
NIce work!

File: node/pkg/accountant/accountant_test.go (hunk 5 above)
Comment (tbjump) [reply node/pkg/accountant/accountant_test.go:135]: hashFromString is only ever called with a single static value, which is expected in this test file, but that's why the linter complains.
